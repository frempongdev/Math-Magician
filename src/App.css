.calc-box {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 15%;
}

.calc-div {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  height: 50%;
  width: 30%;
}

.calc-div > * {
  border: solid 1px rgb(188, 188, 188);
  padding: 2%;
  font-size: 2rem;
  background-color: rgb(243, 243, 243);
}

#answer {
  color: white;
  grid-column: 1 / span 4;
  text-align: end;
  background-color: rgb(146, 146, 146);
}

#zero {
  grid-column: 1 / span 2;
}

.num {
  padding: 0% 4%;
}

.operand {
  padding: 15% 4%;
  background-color: #f4a25f;
}
