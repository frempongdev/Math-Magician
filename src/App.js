import { Route, Routes } from 'react-router';
import './App.css';
import Home from './components/Home';
import Calculator from './components/Calculator';
import Quote from './components/Quote';
import Navigation from './components/Navigation';

function App() {
  return (
    <>
      <Navigation />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/Calculator" element={<Calculator />} />
        <Route path="/Quote" element={<Quote />} />
      </Routes>
    </>
  );
}

export default App;
