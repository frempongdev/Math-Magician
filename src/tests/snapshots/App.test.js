import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../../App.js'; // eslint-disable-line import/extensions

test('Should match App snapshot', () => {
  const tree = renderer.create(

    <BrowserRouter>
      <App />
    </BrowserRouter>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
