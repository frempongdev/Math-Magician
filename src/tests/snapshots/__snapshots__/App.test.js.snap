// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match App snapshot 1`] = `
Array [
  <div
    className="header"
  >
    <div
      className="topicBox"
    >
      <img
        alt="man"
        className="man"
        src="man.png"
      />
      <h1
        className="topic"
      >
        Math Magician
      </h1>
    </div>
    <nav
      className="navBox"
    >
      <ul
        className="nav"
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <span
          className="divider"
        >
          |
        </span>
        <li>
          <a
            href="/Calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <span
          className="divider"
        >
          |
        </span>
        <li>
          <a
            href="/Quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  <div
    className="homeBox"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Mathematics is one of the most fundamental and practical sciences, and it plays an essential role in everyday life. In particular, arithmetic is an essential branch of mathematics that involves fundamental operations like addition, subtraction, multiplication, and division. Arithmetic is ubiquitous in everyday life, from counting money and calculating bills to solving real-world problems. One of the most common tools for arithmetic calculations is the calculator, a device that has become an integral part of our daily routines.
    </p>
    <p>
      With the advent of technology and the internet, calculators are no longer limited to physical devices but can now be accessed on computers and mobile devices through web applications. Math Magician is one such web application that offers a range of arithmetic calculations to users. It is designed to be user-friendly and accessible to people of all ages and levels of mathematical ability. Math Magician offers a simple yet powerful calculator that can perform basic arithmetic calculations, making it an ideal tool for students, teachers, and professionals alike. Whether you need to quickly calculate a tip or split a bill, Math Magician is the go-to tool for all your arithmetic needs.
    </p>
  </div>,
]
`;
